langchain_postgres/v2/async_vectorstore.py

```diff
@@ -557,16 +610,18 @@ async def __query_collection(
         filter_dict = None
         if filter and isinstance(filter, dict):
             safe_filter, filter_dict = self._create_filter_clause(filter)
-        param_filter = f"WHERE {safe_filter}" if safe_filter else ""
+        where_filters = f"WHERE {safe_filter}" if safe_filter else ""
+        and_filters = f"AND ({safe_filter})" if safe_filter else ""
```

Comment:
nit: add a comment on what this is or move this closer to where the code is used.